# 採用

# データ

In [2]:
import csv
titles = []
with open("ted.csv", encoding="utf-8") as f:
  reader = csv.reader(f)
  for row in reader:
    if row not in titles:
      titles.append(row)

titles = sum(titles, []) # 文字列の1次元リストに変換

In [2]:
# csvファイルに追記する関数
def appendcsv(filename, line):
    with open(filename, 'a', newline='\n', encoding='utf-8') as f:
        writer = csv.writer(f)
        writer.writerow(line)

In [4]:
for row in titles:
    print(f"{row}")

第三極を守る緊急要請
トランスジェンダーは恐怖の言葉ではない
歴史上大胆なルイジアナ買収
ブラックホールを破壊できるのか？
世論調査の善し悪し
中国の白蛇伝説とおせっかいな禅師
新しい政治の物語はすべてを変えられる
テンペストを読むべき理由
ニッポン画混ざり合う日本文化を描く絵画
夢応援面談のススメ
映画づくりの秘訣はロマンとソロバン
好奇心を原動力とした研究
私の一部がウニになったわけ
AIによって私達は人間らしさを取り戻せる
カオスを描く天才画家
自分たちで氷河を作る方法
マキャベリアンの本当の意味
水は宝大地は宝人は宝
パンドラの箱
バナナ、そして食料廃棄の醜悪な側面
犬は何故いつまでも生きないのか?
逸話に隠された女性
古い芸術、新しい意匠
未来を変えるためにあなたができること
アイディアが溢れかえる世界における意味あるイノベーション
心から感じるバイオリン
フラナリー・オコナーをなぜ読むべきなのでしょうか？
筋肉疲労の驚くべき原因
新しい言語を学ぶ秘訣
３ヶ月後
健康なうちに死について語ろう
次へ進むものではなく、共に歩んでいくもの
繁栄する持続可能な世界を築く５つの革新的な政策
DNAにデジタルデータを保存する方法
がんと戦うため細胞をバイオハックする
DNAを書き換えて遺伝病を治すことはできるか？
私は死という現実をどう味方にしたか
人種差別の構造を壊す
ビッグデータが脳のミステリーをどのように明らかにするのか
舞台芸術が民主主義に不可欠な理由
身の回りのものすべてがコンピューターになる
反CEOビジネス戦術論
水の利用法
男も女もみんなフェミニストじゃなきゃ
難民が新しい人生を始めるために必要なもの
漫画は教室にふさわしい
シカゴの廃屋をアートに変えた理由
自分らしくあることが私のスーパーパワー
長く忘れられていたウィルスが抗生物質の危機を解決しうるのはなぜか
何が僕に効果的だったか
気候変動に立ち向かえるスーパー植物とは
移民の子供の気持ち
ツイッターが必要としている変革
どうやって難民に精神衛生の支援をもたらすことが出来るか
読書で共感力を身につけよう
読書の力
農作物の40％が捨てられるアメリカの解決策
黒人はこう考える
膵臓がんの新検査方法を開発する
報復ではなく和解を
CRISPRを用いたDNA編集の方法
今こそ自分の態度を振り返る時
カー

In [5]:
print(titles[3220:3225])


['長く続く恋愛関係における欲望の秘訣', '政府による監視', '魅力の科学', '限界で起こる', 'もしこうだったら？をマンガで問う']


# 形態素解析

In [19]:
import MeCab
import re

def Morphological_analysis(row):
    # 名詞・動詞・形容詞のストップワードリスト
    stop_words = [
        # 名詞
        'こと', 'もの','とき', '時', '場合', 'ところ', 'よう', 'これ', 'それ', 'あれ', 'どれ', 'ここ', 'そこ', 'あそこ', 'ため', 'つい', 'わけ', 'TED', 'TEDx', 'TEDxAustin', 'TEDTalks', 'StoryCorpsTEDPrize', 'TEDPrize', 'ただ', 'あて', 'なら', 'すべて', 'うち', 'もと', 'なきゃ', 'みんな', '理由', '自分', '自身', '今', 'いま', '一', '十', '百', '千', '二', '三', '四', '五', '六', '七', '八', '九', '方法', 'D', '法', '事', '何', 'あと', '全て', 'まま', '話', 'なし', 'vs',
        # 動詞
        'する', 'ある', 'いる', 'なる', 'できる', '行う', '思う', '見える', '分かる', 'ない', 'なき', 'なく', 'しよう', '無い', 'し', 'くれる', 'いう', 'さ', 'やっ', 'でき', 'もらう', '見え', 'つけろ', 'みよう', 'い', 'なかろう', 'やり', 'す', '出', 'いけ', 'もっ', '対する', 'よっ', 'とっ', 'なれる', 'なさい', 'やめ', 
        # 形容詞
        'よい', '悪い', '大きい', '小さい', 'すごい', '少ない', '多い', 'いい', '良い'
    ]

    wakati = MeCab.Tagger()

    # 正規表現で不要な文字を削除
    row = re.sub(r'[0-9０-９]+', '', row)
    row = re.sub(r'[!"#$%&\'\\\\()*+,-./:;<=>?@\[\\\]^_`{|}~「」〔〕“”〈〉『』【】＆＊・（）＄＃＠。、？！｀＋￥％]', '', row)

    nodes = wakati.parseToNode(row)
    words = []

    while nodes:
        word = nodes.surface  # 単語
        pos = nodes.feature.split(',')[0]  # 品詞

        # 名詞、動詞、形容詞かつストップワードではない単語のみ抽出
        # if pos in ['名詞', '動詞', '形容詞'] and word not in stop_words:
        
        # 名詞、形容詞かつストップワードではない単語のみ抽出
        if pos in ['名詞', '形容詞'] and word not in stop_words + words:
            words.append(word)
        nodes = nodes.next
    print(words)
    if len(words) > 1:
        return words
    else:
        return ['もう少し長い文章を入力してください。']


In [3]:
import MeCab
import re

# 名詞・動詞・形容詞のストップワードリスト
stop_words = [
    # 名詞
    'こと', 'もの','とき', '時', '場合', 'ところ', 'よう', 'これ', 'それ', 'あれ', 'どれ', 'ここ', 'そこ', 'あそこ', 'ため', 'つい', 'わけ', 'TED', 'TEDx', 'TEDxAustin', 'TEDTalks', 'StoryCorpsTEDPrize', 'TEDPrize', 'ただ', 'あて', 'なら', 'すべて', 'うち', 'もと', 'なきゃ', 'みんな', '理由', '自分', '自身', '今', 'いま', '一', '十', '百', '千', '二', '三', '四', '五', '六', '七', '八', '九', '方法', 'D', '法', '事', '何', 'あと', '全て', 'まま', '話', 'なし', 'vs',
    # 動詞
    'する', 'ある', 'いる', 'なる', 'できる', '行う', '思う', '見える', '分かる', 'ない', 'なき', 'なく', 'しよう', '無い', 'し', 'くれる', 'いう', 'さ', 'やっ', 'でき', 'もらう', '見え', 'つけろ', 'みよう', 'い', 'なかろう', 'やり', 'す', '出', 'いけ', 'もっ', '対する', 'よっ', 'とっ', 'なれる', 'なさい', 'やめ', 
    # 形容詞
    'よい', '悪い', '大きい', '小さい', 'すごい', '少ない', '多い', 'いい', '良い'
]

wakati = MeCab.Tagger()



# corpus = []

for row in titles:

    # 正規表現で不要な文字を削除
    row = re.sub(r'[0-9０-９]+', '', row)
    row = re.sub(r'[!"#$%&\'\\\\()*+,-./:;<=>?@\[\\\]^_`{|}~「」〔〕“”〈〉『』【】＆＊・（）＄＃＠。、？！｀＋￥％]', '', row)

    # MeCabで形態素解析
    nodes = wakati.parseToNode(row)
    words = []
    while nodes:
        word = nodes.surface  # 単語
        pos = nodes.feature.split(',')[0]  # 品詞

        # 名詞、動詞、形容詞かつストップワードではない単語のみ抽出
        # if pos in ['名詞', '動詞', '形容詞'] and word not in stop_words:
        
        # 名詞、形容詞かつストップワードではない単語のみ抽出
        if pos in ['名詞', '形容詞'] and word not in stop_words + words:
            words.append(word)
        nodes = nodes.next

    if len(words) > 2:
        # corpus.append(words)
        appendcsv('ted_corpus.csv', words)
        appendcsv('ted_corpus_label.csv', words)
        print(f"Row: {row}")
        print(f"Words: {words}")



Row: 第三極を守る緊急要請
Words: ['極', '緊急', '要請']
Row: トランスジェンダーは恐怖の言葉ではない
Words: ['トランス', 'ジェンダー', '恐怖', '言葉']
Row: 歴史上大胆なルイジアナ買収
Words: ['歴史', 'ルイジアナ', '買収']
Row: ブラックホールを破壊できるのか
Words: ['ブラック', 'ホール', '破壊']
Row: 世論調査の善し悪し
Words: ['世論', '調査', '善し悪し']
Row: 中国の白蛇伝説とおせっかいな禅師
Words: ['中国', '白蛇', '伝説', 'せっかい', '禅師']
Row: 新しい政治の物語はすべてを変えられる
Words: ['新しい', '政治', '物語']
Row: ニッポン画混ざり合う日本文化を描く絵画
Words: ['ニッポン', '画', '日本', '文化', '絵画']
Row: 夢応援面談のススメ
Words: ['夢', '応援', '面談', 'ススメ']
Row: 映画づくりの秘訣はロマンとソロバン
Words: ['映画', 'づくり', '秘訣', 'ロマン', 'ソロバン']
Row: 好奇心を原動力とした研究
Words: ['好奇', '原動', '研究']
Row: カオスを描く天才画家
Words: ['カオス', '天才', '画家']
Row: マキャベリアンの本当の意味
Words: ['マキャベリ', 'アン', '本当', '意味']
Row: 水は宝大地は宝人は宝
Words: ['水', '宝', '大地']
Row: バナナそして食料廃棄の醜悪な側面
Words: ['バナナ', '食料', '廃棄', '醜悪', '側面']
Row: 古い芸術新しい意匠
Words: ['古い', '芸術', '新しい', '意匠']
Row: アイディアが溢れかえる世界における意味あるイノベーション
Words: ['アイディア', '世界', '意味', 'イノベーション']
Row: 筋肉疲労の驚くべき原因
Words: ['筋肉', '疲労', '原因']
Row: 新しい言語を学ぶ秘訣
Words: ['新しい', '言語', '秘訣']
Row: 繁栄する持続可能な世界を

In [18]:
for row in corpus:
    print(f"{row}")

['三', '極', '緊急', '要請']
['トランス', 'ジェンダー', '恐怖', '言葉']
['歴史', 'ルイジアナ', '買収']
['ブラック', 'ホール', '破壊']
['世論', '調査', '善し悪し']
['中国', '白蛇', '伝説', 'せっかい', '禅師']
['政治', '物語']
['テンペスト', '理由']
['ニッポン', '画', '日本', '文化', '絵画']
['夢', '応援', '面談', 'ススメ']
['映画', 'づくり', '秘訣', 'ロマン', 'ソロバン']
['好奇', '原動', '研究']
['一部', 'ウニ']
['AI', '人間']
['カオス', '天才', '画家']
['自分', '氷河', '方法']
['マキャベリ', 'アン', '本当', '意味']
['水', '宝', '大地', '宝', '宝']
['パンドラ', '箱']
['バナナ', '食料', '廃棄', '醜悪', '側面']
['逸話', '女性']
['芸術', '意匠']
['アイディア', '世界', '意味', 'イノベーション']
['心', 'バイオリン']
['筋肉', '疲労', '原因']
['言語', '秘訣']
['健康', '死']
['次', '共']
['繁栄', '持続', '世界', '革新', '政策']
['DNA', 'デジタル', 'データ', '保存', '方法']
['がん', '細胞', 'バイオハック']
['DNA', '遺伝']
['死', '現実', '味方']
['人種', '差別', '構造']
['データ', '脳', 'ミステリー']
['舞台', '芸術', '民主', '主義', '理由']
['身の回り', 'コンピューター']
['CEO', 'ビジネス', '戦術', '論']
['水', '利用', '法']
['男', '女', 'フェミニスト']
['難民', '人生', '必要']
['漫画', '教室', 'ふさわしい']
['シカゴ', '廃屋', 'アート', '理由']
['自分', 'スーパー', 'パワー']
['長く', 'ウィルス', '抗生', '物質', '危機', '解決']
['気候', 

In [ ]:
corpus = []
with open("ted.csv", encoding="utf-8") as f:
  reader = csv.reader(f)
  for row in reader:
    if row not in corpus:
      corpus.append(row)

In [19]:
print(len(corpus))

3279


# 試し

In [16]:
from sklearn.feature_extraction.text import TfidfVectorizer

vectorizer = TfidfVectorizer(analyzer=lambda x: x)
X = vectorizer.fit_transform(corpus)

# 各文の重要語を抽出
feature_names = vectorizer.get_feature_names_out()
for doc_index, row in enumerate(X):
    sorted_indices = row.toarray().argsort()[0, -2:][::-1]  # スコア上位2語を抽出
    important_words = [feature_names[idx] for idx in sorted_indices]
    print(f"文 {doc_index + 1} の重要語: {important_words}")


文 1 の重要語: ['要請', '極']
文 2 の重要語: ['トランス', 'ジェンダー']
文 3 の重要語: ['買収', 'ルイジアナ']
文 4 の重要語: ['破壊', 'ホール']
文 5 の重要語: ['世論', '善し悪し']
文 6 の重要語: ['禅師', '白蛇']
文 7 の重要語: ['政治', '物語']
文 8 の重要語: ['テンペスト', '理由']
文 9 の重要語: ['画', 'ニッポン']
文 10 の重要語: ['面談', '応援']
文 11 の重要語: ['ソロバン', 'ロマン']
文 12 の重要語: ['原動', '好奇']
文 13 の重要語: ['ウニ', '一部']
文 14 の重要語: ['AI', '人間']
文 15 の重要語: ['画家', 'カオス']
文 16 の重要語: ['氷河', '自分']
文 17 の重要語: ['マキャベリ', 'アン']
文 18 の重要語: ['宝', '大地']
文 19 の重要語: ['パンドラ', '箱']
文 20 の重要語: ['醜悪', 'バナナ']
文 21 の重要語: ['逸話', '女性']
文 22 の重要語: ['意匠', '芸術']
文 23 の重要語: ['イノベーション', 'アイディア']
文 24 の重要語: ['バイオリン', '心']
文 25 の重要語: ['疲労', '筋肉']
文 26 の重要語: ['秘訣', '言語']
文 27 の重要語: ['死', '健康']
文 28 の重要語: ['共', '次']
文 29 の重要語: ['繁栄', '政策']
文 30 の重要語: ['保存', 'デジタル']
文 31 の重要語: ['バイオハック', 'がん']
文 32 の重要語: ['遺伝', 'DNA']
文 33 の重要語: ['味方', '死']
文 34 の重要語: ['構造', '差別']
文 35 の重要語: ['ミステリー', 'データ']
文 36 の重要語: ['舞台', '民主']
文 37 の重要語: ['身の回り', 'コンピューター']
文 38 の重要語: ['CEO', '戦術']
文 39 の重要語: ['利用', '水']
文 40 の重要語: ['女', '男']
文 41 

# 学習

In [4]:
list = ['あ', 0, 1]
print(list[:-2])

['あ']


In [11]:
from sklearn.feature_extraction.text import TfidfVectorizer

def extract_important_words(corpus):
    vectorizer = TfidfVectorizer(analyzer=lambda x: x)  # 単語のリストをそのまま渡す
    X_vec = vectorizer.fit_transform(corpus)

    # 各文の重要語を抽出
    feature_names = vectorizer.get_feature_names_out()

    sorted_indices = X_vec[len(X)-1].toarray().argsort()[0, -2:][::-1]  # スコア上位2語を抽出
    important_words = [feature_names[idx] for idx in sorted_indices]

    return important_words

# main

In [12]:
import csv

X = []
y = []

with open("ted_corpus_label.csv", encoding="utf-8") as f:
  reader = csv.reader(f)
  for row in reader:
      y.append([int(row[-2]), int(row[-1])])
      X.append(row[:-2])

# X = [" ".join(words) for words in X]


print(len(y), y)
print(len(X), X)

2137 [[1, 2], [0, 1], [1, 2], [0, 1], [0, 1], [0, 1], [0, 1], [2, 4], [0, 1], [0, 2], [0, 2], [0, 2], [0, 1], [0, 2], [0, 2], [1, 3], [0, 1], [0, 1], [1, 2], [0, 3], [0, 1], [0, 2], [0, 1], [0, 1], [0, 1], [0, 2], [0, 1], [0, 1], [1, 3], [0, 2], [0, 2], [1, 5], [0, 3], [0, 1], [0, 2], [0, 3], [0, 1], [0, 3], [0, 2], [0, 1], [0, 2], [0, 2], [0, 1], [1, 3], [0, 1], [2, 3], [0, 2], [0, 2], [0, 4], [0, 1], [0, 2], [0, 2], [2, 4], [1, 2], [0, 1], [0, 3], [0, 3], [1, 2], [0, 4], [0, 2], [0, 2], [0, 3], [1, 2], [0, 1], [0, 2], [0, 3], [1, 5], [0, 2], [0, 2], [0, 2], [1, 2], [2, 4], [0, 2], [0, 1], [0, 1], [0, 2], [0, 1], [1, 3], [0, 1], [0, 2], [0, 1], [3, 4], [0, 1], [2, 3], [0, 1], [0, 1], [1, 2], [0, 3], [0, 3], [0, 2], [0, 1], [1, 2], [0, 1], [0, 4], [0, 1], [0, 1], [1, 2], [0, 1], [0, 3], [1, 2], [0, 1], [0, 1], [1, 2], [0, 2], [0, 1], [0, 1], [0, 1], [0, 1], [0, 3], [1, 4], [0, 1], [0, 1], [1, 5], [0, 1], [1, 2], [1, 2], [0, 2], [0, 3], [0, 2], [1, 3], [0, 3], [0, 1], [0, 1], [0, 2], [1

In [22]:
# 文章を形態素解析して処理結果を返す
input_word = '発表の内容からスライドの配色を提案する'

# 形態素解析で得られた単語リストをコーパスに追加
processed_word = Morphological_analysis(input_word)

# コーパスに追加された形態素解析の結果を使って重要語を抽出
X = [processed_word]  # 他の文もあれば追加
important_words = extract_important_words(X)

# 結果を表示
for result in important_words:
    print(result)


['発表', '内容', 'スライド', '配色', '提案']
配色
発表


# スコア
**uni-gram**  


In [69]:
from sklearn.feature_extraction.text import TfidfVectorizer

vectorizer = TfidfVectorizer(analyzer=lambda x: x)
X_vec = vectorizer.fit_transform(X)

# print(X)

# 各文の重要語を抽出
feature_names = vectorizer.get_feature_names_out()
for doc_index, row in enumerate(X_vec):
    sorted_indices = row.toarray().argsort()[0, -2:][::-1]  # スコア上位2語を抽出
    predicted_words = [feature_names[idx] for idx in sorted_indices]
    print(f"文 {doc_index + 1} の重要語: {predicted_words}")

文 1 の重要語: ['極', '要請']
文 2 の重要語: ['トランス', '恐怖']
文 3 の重要語: ['ルイジアナ', '買収']
文 4 の重要語: ['ブラックホール', '破壊']
文 5 の重要語: ['世論', '善し悪し']
文 6 の重要語: ['白蛇', '禅師']
文 7 の重要語: ['政治', '物語']
文 8 の重要語: ['画', 'ニッポン']
文 9 の重要語: ['面談', 'ススメ']
文 10 の重要語: ['ソロバン', 'ロマン']
文 11 の重要語: ['原動', '好奇']
文 12 の重要語: ['画家', 'カオス']
文 13 の重要語: ['アン', 'マキャベリ']
文 14 の重要語: ['大地', '宝']
文 15 の重要語: ['醜悪', '側面']
文 16 の重要語: ['意匠', '古い']
文 17 の重要語: ['アイディア', 'イノベーション']
文 18 の重要語: ['筋肉', '疲労']
文 19 の重要語: ['言語', '秘訣']
文 20 の重要語: ['繁栄', '政策']
文 21 の重要語: ['保存', 'デジタル']
文 22 の重要語: ['バイオハック', 'がん']
文 23 の重要語: ['味方', '死']
文 24 の重要語: ['構造', '差別']
文 25 の重要語: ['ミステリー', 'データ']
文 26 の重要語: ['舞台', '民主']
文 27 の重要語: ['戦術', 'CEO']
文 28 の重要語: ['フェミニスト', '男']
文 29 の重要語: ['難民', '新しい']
文 30 の重要語: ['漫画', 'ふさわしい']
文 31 の重要語: ['シカゴ', '廃屋']
文 32 の重要語: ['ウィルス', '長く']
文 33 の重要語: ['スーパー', '植物']
文 34 の重要語: ['移民', '気持ち']
文 35 の重要語: ['ツイッター', '変革']
文 36 の重要語: ['衛生', '支援']
文 37 の重要語: ['共感', '読書']
文 38 の重要語: ['農作', '策']
文 39 の重要語: ['膵臓', '検査']
文 40 の重要語: ['CRISPR',

In [71]:
from sklearn.metrics import precision_score, recall_score, f1_score

# 評価用
precision_list = []
recall_list = []
f1_list = []

for doc_index, row in enumerate(X_vec):
    # 予測重要語: X の中から上位 2 単語（ただしインデックスで指定）
    sorted_indices = row.toarray().argsort()[0, -2:][::-1]  # スコア上位2語を抽出
    predicted_words = [feature_names[idx] for idx in sorted_indices]

    # 正解ラベル（重要語のインデックス）
    true_indices = y[doc_index]
    # print(true_indices)
    # print(X[doc_index])
    # doc_length = len(X[doc_index])
    # for i in true_indices:
    #     if i >= doc_length:
    #         print(f"警告{doc_index}: インデックス {i} は、長さ {doc_length} のドキュメントの範囲外です")
    
    # 正解ラベル（重要語）
    true_words = [X[doc_index][i] for i in true_indices]

    # Precision, Recall, F1の計算
    tp = len(set(predicted_words) & set(true_words))
    fp = len(set(predicted_words) - set(true_words))
    fn = len(set(true_words) - set(predicted_words))

    precision = tp / (tp + fp) if (tp + fp) > 0 else 0
    recall = tp / (tp + fn) if (tp + fn) > 0 else 0
    f1 = (2 * tp) / (2 * tp + fp + fn) if (2 * tp + fp + fn) > 0 else 0

    precision_list.append(precision)
    recall_list.append(recall)
    f1_list.append(f1)

    print(f"文 {doc_index + 1} の予測重要語: {predicted_words}")
    print(f"文 {doc_index + 1} の正解重要語: {true_words}")
    print(f"Precision: {precision}  Recall: {recall}  F1: {f1}")

# スコアの平均値を出力
print(f"平均 Precision: {sum(precision_list) / len(precision_list):.4f}")
print(f"平均 Recall: {sum(recall_list) / len(recall_list):.4f}")
print(f"平均 F1スコア: {sum(f1_list) / len(f1_list):.4f}")


文 1 の予測重要語: ['極', '要請']
文 1 の正解重要語: ['緊急', '要請']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 2 の予測重要語: ['トランス', '恐怖']
文 2 の正解重要語: ['トランス', 'ジェンダー']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 3 の予測重要語: ['ルイジアナ', '買収']
文 3 の正解重要語: ['ルイジアナ', '買収']
Precision: 1.0  Recall: 1.0  F1: 1.0
文 4 の予測重要語: ['ブラックホール', '破壊']
文 4 の正解重要語: ['ブラックホール', '破壊']
Precision: 1.0  Recall: 1.0  F1: 1.0
文 5 の予測重要語: ['世論', '善し悪し']
文 5 の正解重要語: ['世論', '調査']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 6 の予測重要語: ['白蛇', '禅師']
文 6 の正解重要語: ['中国', '白蛇']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 7 の予測重要語: ['政治', '物語']
文 7 の正解重要語: ['新しい', '政治']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 8 の予測重要語: ['画', 'ニッポン']
文 8 の正解重要語: ['日本', '絵画']
Precision: 0.0  Recall: 0.0  F1: 0.0
文 9 の予測重要語: ['面談', 'ススメ']
文 9 の正解重要語: ['夢', '応援']
Precision: 0.0  Recall: 0.0  F1: 0.0
文 10 の予測重要語: ['ソロバン', 'ロマン']
文 10 の正解重要語: ['映画', '秘訣']
Precision: 0.0  Recall: 0.0  F1: 0.0
文 11 の予測重要語: ['原動', '好奇']
文 11 の正解重要語: ['好奇', '研究']
Precision: 0.5  Recall: 0.5  F1: 0.5
文 12 の予測重要

# データ
- 日本語翻訳されたTEDのタイトルを収集
- 発表者名はプログラムと人手で取り除いた
- その他、一語のタイトルや上手く解析できないタイトルは取り除いた
- 数字と記号を削除
- 名詞と形容詞を抽出
- 2語以上のものをコーパスに格納

# 課題
- 人名が（二語に分かれて）抽出されがち
- デフォルトの辞書に登録されている単語数が少ない（'トランスジェンダー'が'トランス'と'ジェンダー'）
- 数字をすべて削除すると、'3Dプリンター'などの単語が抽出されない
- データ数が少ない(3279)

# 評価
- 適合率と再現率が同じ。なのでF値も同じ値になる
- 平均 Precision: 0.5962
- 平均 Recall: 0.5962
- 平均 F1スコア: 0.5962
